### PR TITLE
@kanaabe => Text component / Paragraph updates

### PR DESCRIPTION
**Pull request by eessex**
- Adds Reaction Text component to Text Section
- Change import style for Draft imports
- Replace 'rich_text2/input_paragraph' with 'rich_text2/paragraph' (will also replace rich_text_caption in a separate PR)
- Changes sectionContainer to mimic Reaction component naming
- Kills [handleBackspace error](https://sentry.io/artsynet/positron-staging/issues/330131418/)

### CODE REVIEW

File: client/apps/edit/components/content2/section_container/index.coffee [hunk 1]
```diff
@@ -6,14 +6,13 @@
...
-SectionCallout = React.createFactory require '../../content/sections/callout/index.coffee'
-SectionImageCollection = React.createFactory require '../sections/image_collection/index.coffee'
-SectionImage = React.createFactory require '../../content/sections/image/index.coffee'
+Text = React.createFactory require '../sections/text/index.coffee'
+Video = React.createFactory require '../../content/sections/video/index.coffee'
+Slideshow = React.createFactory require '../../content/sections/slideshow/index.coffee'
+Embed = React.createFactory require '../../content/sections/embed/index.coffee'
+Fullscreen = React.createFactory require '../../content/sections/fullscreen/index.coffee'
+ImageCollection = React.createFactory require '../sections/image_collection/index.coffee'
+Image = React.createFactory require '../../content/sections/image/index.coffee'
```
Comment (kanaabe) [inline client/apps/edit/components/content2/section_container/index.coffee]: Nice renaming

File: client/apps/edit/components/content2/sections/text/index.coffee [hunk 2]
```diff
@@ -3,39 +3,27 @@ ReactDOM = require 'react-dom'
 _s = require 'underscore.string'
 sd = require('sharify').data
 window.process = {env: {NODE_ENV: sd.NODE_ENV}}
+components = require('@artsy/reaction-force/dist/components/publishing/index').default
+Config = require './draft_config.coffee'
+Nav = React.createFactory require '../../../../../../components/rich_text2/components/nav.coffee'
+InputUrl = React.createFactory require '../../../../../../components/rich_text2/components/input_url.coffee'
+Text = React.createFactory components.Text
+Utils = require '../../../../../../components/rich_text2/utils/index.coffee'
```
Comment (kanaabe) [inline client/apps/edit/components/content2/sections/text/index.coffee]: 👍

File: client/apps/edit/components/content2/sections/text/index.styl [hunk 3]
```diff
@@ -29,89 +27,6 @@
...
-      border-radius 50%
-      color rgba(0,0,0,0)
-  .content-start
-    Unica s69, medium
-    float left
-    line-height .5em
-    margin-right 10px
-    margin-top .298em
-    text-transform uppercase
-
```
Comment (kanaabe) [inline client/apps/edit/components/content2/sections/text/index.styl]: 👏

File: package.json [hunk 4]
```diff
@@ -14,7 +14,7 @@
     "deploy": "sh scripts/deploy.sh"
   },
   "dependencies": {
-    "@artsy/reaction-force": "^0.1.49",
+    "@artsy/reaction-force": "^0.3.1",
```
Comment (kanaabe) [inline package.json]: Not sure if you want to include any of the latest Reaction work here, but I imagine we should be at a 0.4.0 now if you want it.

Comment (eessex) [inline package.json]: Ah cool, yeah definitely want the latest

File: client/apps/edit/components/content2/sections/text/index.coffee [hunk 5]
```diff
@@ -93,7 +93,7 @@ module.exports = React.createClass
     selection = Utils.getSelectionDetails(@state.editorState)
     # only merge a section if cursor is in first character of first block
     if selection.isFirstBlock and selection.anchorOffset is 0 and
-     @props.sections.models[@props.index - 1].get('type') is 'text'
+     @props.sections.models[@props.index - 1]?.get('type') is 'text'
```
Comment (kanaabe) [inline client/apps/edit/components/content2/sections/text/index.coffee]: Check out `.indexOf(@props.index -1)` or `.at(@props.index -1)`

**Comment by kanaabe** [comment]
I can't merge because of the conflict, but this is ready otherwise. Nice job refactoring the text component to use Reaction! It's a very ideal outcome :)

File: package.json (hunk 4 above)
Comment (eessex) [inline package.json]: I'm going to go ahead and deploy reaction -- is still on 3.1, do i jump right to v4.0?

**Pull request merged**